core/src/main/java/org/apache/iceberg/AllManifestsTable.java

```diff
@@ -168,31 +153,34 @@ static class ManifestListReadTask implements DataTask {
     private final FileIO io;
     private final Schema schema;
     private final Map<Integer, PartitionSpec> specs;
-    private final FileScanTask manifestListTask;
+    private final String manifestListLocation;
+    private final Expression residual;
     private final long referenceSnapshotId;
 
     ManifestListReadTask(
         FileIO io,
         Schema schema,
         Map<Integer, PartitionSpec> specs,
-        FileScanTask manifestListTask,
+        String manifestListLocation,
+        Expression residual,
         long referenceSnapshotId) {
       this.io = io;
       this.schema = schema;
       this.specs = specs;
-      this.manifestListTask = manifestListTask;
+      this.manifestListLocation = manifestListLocation;
+      this.residual = residual;
       this.referenceSnapshotId = referenceSnapshotId;
     }
 
     @Override
     public List<DeleteFile> deletes() {
-      return manifestListTask.deletes();
+      return ImmutableList.of();
```

Comment:
Because we don't have any possible delete files here since we are just reading for manifests?

Reply:
That's right. We don't have delete files against manifests.